runtime/permissions.rs

```diff
@@ -57,14 +57,18 @@ impl<TSys: RuntimePermissionDescriptorParserSys + std::fmt::Debug>
     &self,
     text: &str,
   ) -> Result<ReadDescriptor, PathResolveError> {
-    Ok(ReadDescriptor(self.parse_path_query(text)?))
+    Ok(ReadDescriptor(
+      self.parse_path_query(Cow::Borrowed(Path::new(text)))?,
```

Comment:
Slightly more verbose in this case, but it's forcing the caller to think about this.